locator/tablets.hh

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "dht/token.hh"
+#include "utils/small_vector.hh"
+#include "locator/host_id.hh"
+#include "dht/i_partitioner_fwd.hh"
+#include "schema/schema_fwd.hh"
+#include "utils/chunked_vector.hh"
+#include "utils/hash.hh"
+
+#include <seastar/core/reactor.hh>
+#include <seastar/util/log.hh>
+
+#include <vector>
+
+namespace locator {
+
+extern seastar::logger tablet_logger;
+
+using token = dht::token;
+
+// Identifies tablet within the scope of a single tablet_map,
+// which has a scope of (table_id, token metadata version).
+// Different tablets of different tables can have the same tablet_id.
+// Different tablets in subsequent token metadata version can have the same tablet_id.
+// When splitting a tablet, one of the new tablets (in the new token metadata version)
+// will have the same tablet_id as the old one.
+using tablet_id = size_t;
```

Comment:
nit: typically, we use `_id` for unique uuid-based identifiers.
I think `table_idx` would better describe what it stands for.

Reply:
Also, it should be a tagged integer as proposed in https://github.com/scylladb/scylladb/pull/12966/commits/abb832238c015ab7428d95d363f058a923c7e352

Reply:
We also have shard_id.

Reply:
There's `enum class : size_t`, no need for tagged_integral.

Reply:
> We also have shard_id.

I think we would eventually also want to identify tablet replicas globally by a uuid,
so I think the in-memory tablet_id would be confusing for that reason.

Reply:
> I think we would eventually also want to identify tablet replicas globally by a uuid,
> so I think the in-memory tablet_id would be confusing for that reason.

We can introduce tablet_replica_id then. I think it's a matter of terminology, what a "tablet" is. It could be also confusing to call the tablet replica by "tablet_id" if you don't know what "tablet" refers to. I propose to stick to the terminology where "tablet" is the thing at the top of the hierarchy, unless the context says otherwise. If there is no context, we use "tablet replica" for tablet replica.

Reply:
okay :crossed_fingers: